edit/edit_test.go

```diff
@@ -698,6 +709,18 @@ func TestEditorEditSubstitute(t *testing.T) {
 			{edit: "a/abc", want: "abc"},
 			{edit: `s/abc/\1`, want: ""},
 		},
+		{
+			{edit: "a/abc abc abc", want: "abc abc abc"},
+			{edit: "s0/abc/hello/", want: "hello abc abc"},
+		},
+		{
+			{edit: "a/abc abc abc", want: "abc abc abc"},
+			{edit: "s2/abc/hello/", want: "abc hello abc"},
+		},
+		{
+			{edit: "a/abc abc abc", want: "abc abc abc"},
+			{edit: "s2/abc/hello/g", want: "abc hello hello"},
+		},
```

Comment:
Added n == 11 test cases (both with and without g), as well as those listed above.